app/src/main/java/org/wikipedia/page/PageActivity.java

```diff
@@ -468,6 +471,20 @@ public void loadMainPage(TabPosition position) {
         loadPage(title, historyEntry, position);
     }
 
+    private void loadFilePageFromBackStackIfNeeded() {
```

Comment:
Lgtm. Just one minor question..How do we trigger this? when will a filepage be in the backstack?

Reply:
If the tab count = 1 and tap on the tab view (the one with currently opened tabs), it will take you directly to the page.